src/core/shards/wires.cpp

```diff
@@ -165,7 +166,7 @@ struct WireBase {
     } else {
       SHLOG_TRACE("Skipping {} compose", wire->name);
       // verify input type
-      if (!passthrough && mode != Stepped && data.inputType != wire->inputType) {
+      if (!passthrough && mode != Stepped && data.inputType != wire->inputType && !wire->inputTypeForceNone) {
```

Comment:
@guusw check this for why `inputTypeForceNone` is needed. As None could be legit and not match.